go/vt/vtgate/evalengine/arithmetic.go

```diff
@@ -158,120 +148,175 @@ func makeNumericAndPrioritize(left, right eval) (evalNumeric, evalNumeric) {
 	return i1, i2
 }
 
-func intPlusIntWithError(v1, v2 int64) (eval, error) {
+func mathAdd_ii(v1, v2 int64) (eval, error) {
+	result, err := mathAdd_ii0(v1, v2)
+	return newEvalInt64(result), err
```

Comment:
I guess we're ok with the small difference here that we do return a `newEvalInt64` value on error now (with the value `0`) vs. returning `nil` before? 

It does add an allocation in the error path, but that's then rare enough?